examples/flutter_gallery/lib/demo/animation_demo.dart

```diff
@@ -272,12 +276,21 @@ class _RectangleDemoState extends State<_RectangleDemo> {
 
   CurvedAnimation _animation;
   _DragTarget _dragTarget;
-  Rect _begin = new Rect.fromLTRB(180.0, 100.0, 330.0, 200.0);
-  Rect _end = new Rect.fromLTRB(32.0, 275.0, 132.0, 425.0);
+  Rect _begin;
+  Rect _end;
 
   @override
   void initState() {
     super.initState();
+    final Size screenSize = MediaQuery.of(context).size;
```

Comment:
I think the current code is fine. Other ways you could do it are to use fractions in the animations, and paint using those in the custom painter (by multiplying by the canvas size). Generally I like to do things that way because then if you put the same canvas inside something else, it keeps working (whereas the media query version only works when it's full-screen). But since this is a concrete demo and not a widget, reuse is not as important.